addons/account/tests/test_reconciliation_matching_rules.py

```diff
@@ -337,8 +337,8 @@ def test_matching_fields_match_text_location_no_partner(self):
             }, statements=self.bank_st)
 
             # Test matching with the partner name (reinitializing the statement line first)
-            self.bank_line_1.write({**st_line_initial_vals, st_line_field: self.partner_1.name})
-
+            self.bank_line_1.write({**st_line_initial_vals})
+            self.bank_line_1[st_line_field] = self.partner_1.name
```

Comment:
What's behind this?

Reply:
Function _check_statement_matching() calls function _apply_rules()
Function _apply_rules() calls flush()
Function flush() calls function recompute()

Reply:
> When `st_line_field` equal `narration`, you will write two times with two different values

What? That's not how dict comprehension works, the last value is kept when building the dict. So the `write` function only sees one value.
Now, you are doing twice a `write` because of this change (the explicit one and an implicit one)

Reply:
When writing on payment_ref, a write on partner_id is made in function _synchronize_to_moves:

https://github.com/odoo/odoo/blob/15.0/addons/account/models/account_bank_statement.py#L1006

In the write function, function modified is called

https://github.com/odoo/odoo/blob/15.0/odoo/models.py#L3850

And it will add the field narration in self.all.tocompute

https://github.com/odoo/odoo/blob/15.0/odoo/api.py#L758

Meaning that the next time it reads the field narration, the function _compute_narration will be called.
So it will reset narration to False